/Logger for DB connection

miabu-pashh opened the pull request:
# Pull Request Template

**Fixes #2 r**

### What was changed?
This update implements logging for the database connection in the Donor form file. It includes detailed logging of the Prisma database connection status in the console, enabling better monitoring during application startup.

### Notable Changes:

Added logging functionality for the Prisma connection in index.ts.
This ensures the connection to the database is logged in the console, allowing real-time monitoring during application startup.

### Why was it changed?

This enhancement improves the visibility of the database connection process during the build and runtime phases of the application. By logging the Prisma connection details, we can ensure that automatic migrations are successfully applied and that the database is connected and running properly on the backend.

This is crucial for identifying and diagnosing connection issues early, especially in production environments.


### How was it changed?

The database connection function has been integrated into the index.ts file, where it logs the connection status to the console. This is triggered whenever the application starts, providing real-time feedback on the status of the Prisma connection.

Key updates include:

Logging Prisma connection details during application startup.
Ensuring that any issues with database connectivity are immediately visible via the console.


### Screenshots that show the changes (if applicable):

<img width="1413" alt="image" src="https://github.com/user-attachments/assets/2a437fea-fa6b-4b19-a383-8c4b4295d56f">

____

<img width="1239" alt="image" src="https://github.com/user-attachments/assets/73e66d3b-500b-4050-93ed-9db2c2b2d563">

Anjali0407-git commented:
@miabu-pashh Changes looks good. I will test it and let you know

saikiran0405 commented on server/src/index.ts:
```diff
@@ -5,6 +5,9 @@ import cookieParser from 'cookie-parser';
+import { PrismaClient } from '@prisma/client'; // Import Prisma
+const prisma = new PrismaClient(); // Initialize Prisma Client
+
 // import indexRouter from './routes/index';
```
It's good to remove unwanted code

saikiran0405 commented on server/src/index.ts:
```diff
@@ -47,9 +51,35 @@ app.use((err: any, req: Request, res: Response, next: NextFunction) => {
+// });
+
+// Logging the DB connection and starting server
+const startServer = async () => {
```
In your catch block, you are logging generic error messages.
Possible solution: We can add more context to the error logs, such as timestamps or details about which part of the code failed.

saikiran0405 commented on server/src/index.ts:
```diff
@@ -47,9 +51,35 @@ app.use((err: any, req: Request, res: Response, next: NextFunction) => {
+    app.listen(port, () => {
+      console.log(`Server running on http://localhost:${port}`);
+    });
+  } catch (error) {
```
In your catch block, you are logging generic error messages.
Possible solution: We can add more context to the error logs, such as timestamps or details about which part of the code failed.

Anjali0407-git commented:
@miabu-pashh I have finished testing your update and it correctly logs db connection status to the user. I have requested few suggestions to work on.

kate-holdener commented on client-app/src/Components/Login.js:
Looks like you didn't make any meaningful changes in this file (add a blank line/remove a blank line). Please don't commit files that didn't need to be changed.

miabu-pashh replied:
Yeah, understood. Thanks for the update.

kate-holdener commented on server/src/index.ts:
```diff
@@ -47,9 +51,35 @@ app.use((err: any, req: Request, res: Response, next: NextFunction) => {
+// const port = process.env.PORT || 3001; // Use environment variable or default to 3001
+// app.listen(port, () => {
+//   console.log(`Server running on http://localhost:${port}`);
+// });
```
Please don't leave commented out code in your final solution. Remove the code that's not needed.

miabu-pashh replied:
yes, thank you. This code smells, I haven't taken it under consideration while pushing the code, It was before our recent class. Thanks for the update The next pr will not contain such activity.

kate-holdener commented on server/src/routes/donorRoutes.ts:
Looks like you didn't really change this file (added a blank line). Please don't commit files that you didn't intentionally modify.

miabu-pashh commented:
I have implemented the changes that have been discussed and updated the code.

<img width="1470" alt="image" src="https://github.com/user-attachments/assets/f9127693-f7e7-43cf-9899-6b86d1b9e229">

Anjali0407-git commented:
@miabu-pashh some review comments were left unaddressed. Please address them
